modules/core/src/v2/keychains.ts

```diff
@@ -120,46 +128,38 @@ Keychains.prototype.updatePassword = function(params, callback) {
  * @param callback
  * @returns {object}
  */
-Keychains.prototype.updateSingleKeychainPassword = function({ keychain, oldPassword, newPassword }) {
-  if (!_.isString(oldPassword)) {
+Keychains.prototype.updateSingleKeychainPassword = function( params: { keychain: any, oldPassword: string, newPassword: string }): any {
+
+  if (!_.isString(params.oldPassword)) {
     throw new Error('expected old password to be a string');
   }
 
-  if (!_.isString(newPassword)) {
+  if (!_.isString(params.newPassword)) {
     throw new Error('expected new password to be a string');
   }
 
-  if (!_.isObject(keychain) || !_.isString(keychain.encryptedPrv)) {
+  if (!_.isObject(params.keychain) || !_.isString(params.keychain.encryptedPrv)) {
     throw new Error('expected keychain to be an object with an encryptedPrv property');
   }
 
-  const oldEncryptedPrv = keychain.encryptedPrv;
+  const oldEncryptedPrv = params.keychain.encryptedPrv;
   try {
-    const decryptedPrv = this.bitgo.decrypt({ input: oldEncryptedPrv, password: oldPassword });
-    const newEncryptedPrv = this.bitgo.encrypt({ input: decryptedPrv, password: newPassword });
-    return _.assign({}, keychain, { encryptedPrv: newEncryptedPrv });
+    const decryptedPrv = this.bitgo.decrypt({ input: oldEncryptedPrv, password: params.oldPassword });
+    const newEncryptedPrv = this.bitgo.encrypt({ input: decryptedPrv, password: params.newPassword });
+    return _.assign({}, params.keychain, { encryptedPrv: newEncryptedPrv });
   } catch (e) {
     // catching an error here means that the password was incorrect or, less likely, the input to decrypt is corrupted
     throw new Error('password used to decrypt keychain private key is incorrect');
   }
 };
 
-Keychains.prototype.create = function(params) {
+Keychains.prototype.create = function(params: { seed?: string }): KeyPair {
   params = params || {};
   common.validateParams(params, [], []);
-
   return this.baseCoin.generateKeyPair(params.seed);
 };
 
-Keychains.prototype.add = function(params, callback) {
-  params = params || {};
-  common.validateParams(params, [], ['pub', 'encryptedPrv', 'type', 'source', 'originalPasscodeEncryptionCode', 'enterprise', 'derivedFromParentWithSeed'], callback);
-
-  if (!_.isUndefined(params.disableKRSEmail)) {
```

Comment:
We still need this - please don't remove. Typescript does not validate the objects at runtime.